### PR TITLE
Fix ad slot indexing on DCR Fronts

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
There's a bug in the indexing of ad slots, they are taking the index of the section that they are in/adjacent to rather than the index of the ads on the page.

This PR alter the ad slot indexing so that the id, class etc. are correct.

## Why?
Gain parity with frontend, the inconsistent indexing also makes targeting hard for some campaigns.

### CODE REVIEW

**Comment by ioannakok** [comment]
We crossed wires I think! https://github.com/guardian/dotcom-rendering/pull/7976

**Comment by Jakeii** [comment]
@ioannakok 😵‍💫

File: dotcom-rendering/src/components/DynamicSlowMPU.tsx [hunk 1]
```diff
@@ -283,12 +283,12 @@ type NonMPUProps = {
 const NonMPUSlice = ({ trails, containerPalette, showAge }: NonMPUProps) => {
 	return (
 		<UL direction="row" padBottom={true}>
-			{trails.slice(0, 4).map((card, cardIndex) => {
+			{trails.slice(0, 4).map((card, cardadIndex) => {
```
Comment (ioannakok) [inline dotcom-rendering/src/components/DynamicSlowMPU.tsx:286]:
```suggestion
			{trails.slice(0, 4).map((card, cardAdIndex) => {
```

Comment (ioannakok) [inline dotcom-rendering/src/components/DynamicSlowMPU.tsx:286]: I think this can stay `cardIndex` as it's inside the `NonMPUSlice`

Comment (Jakeii) [reply dotcom-rendering/src/components/DynamicSlowMPU.tsx:286]: Thanks! find and replace gone astray

**Review by ashishpuliyel — APPROVED** [review]
💙 to see it. @SiAdcock and I looked at this yesterday, and agreed we needed a professional to sort it out.

File: dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx [hunk 2]
```diff
@@ -106,15 +106,15 @@ const ThreeColumnSliceWithAdSlot = ({
 				 * |_______________________|
 				 */}
 				<UL direction="row" wrapCards={true}>
-					{trails.map((trail, trailIndex, { length }) => (
+					{trails.map((trail, trailadIndex, { length }) => (
```
Comment (mxdvl) [inline dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx:109]: What’s trailad 😉 – even urbandictionary.com/trailad is of no help

Comment (Jakeii) [reply dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx:109]: oops! 🙊

**Pull request merged**